### PR TITLE
API MFA is always on, and cannot be disabled

### LINKED ISSUE #144: MFA is optional by default
As a CMS owner, I don't want the ability to completely remove MFA once it's been enabled, so that I know the site will stay secure and user preferences for MFA are still respected.

AC
- Once the module is installed, MFA will be optional be default
- It is not possible through the CMS to entirely disable MFA on a site 

Notes:
- Checkbox is removed, radio button is toggled between 'optional' and 'enforced', default on optional

### PR DESCRIPTION
**Pull request by NightJar**
MFA once enabled (i.e. installed) should not be able to be universally
disabled by an administrator. They can alter the "required-ness" of MFA,
i.e. whether or not users must register a second factor method or not,
but only users can add and remove registered methods from their account.
Once set up, MFA is always protecting that that account, whether
required by the site or not.

Closes #144 

image

### CODE REVIEW

**Comment by newleeland** [comment]
Hey @NightJar looking good. I've got a just text changes to make this clearer.

**Text changes**
59318341-cf69e300-8d1a-11e9-8e02-de036fafe076-2

Also, is there a reason why the radio labels are not aligned with radio?

File: client/src/legacy/SiteConfig.js [hunk 1]
```diff
@@ -1,45 +1,14 @@
-/* global window */
-const hiddenClass = 'mfa-settings--hidden';
```
Comment (robbieaverill) [inline client/src/legacy/SiteConfig.js]: Is there some CSS to remove here too?

File: src/Extension/SiteConfigExtension.php [hunk 2]
```diff
@@ -43,30 +41,27 @@ public function updateCMSFields(FieldList $fields)
...
-        $mfaOptions->addExtraClass('mfa-settings__required mfa-settings--hidden');
+        $mfaOptions->addExtraClass('mfa-settings__required');
 
         $mfaGraceEnd = DateField::create(
             'MFAGracePeriodExpires',
             'Optional: grace period end date when MFA is enforced'
         );
+        $mfaGraceEnd->setDescription(_t(
+            __CLASS__ . '.MFA_GRACE_DESCRIPTION',
+            'MFA setup will be optional during this grace period, and reminders to set up will appear at log in.'
```
Comment (robbieaverill) [inline src/Extension/SiteConfigExtension.php]: Possible improvement: "and reminders to register will appear at log in."

File: client/src/legacy/SiteConfig.js (hunk 1 above)
Comment (NightJar) [reply client/src/legacy/SiteConfig.js]: I believe I've nabbed it already (in `SiteConfigExtension.php`), but can search to be sure that's all of it :)

Comment (robbieaverill) [reply client/src/legacy/SiteConfig.js]: See SiteConfig.scss in the legacy folder

File: src/Extension/SiteConfigExtension.php (hunk 2 above)
Comment (NightJar) [reply src/Extension/SiteConfigExtension.php]: I was thinking that as I typed it, but was also aware that changes were inbound from PUX, so didn't think about it too much. The more one thinks about it the more and more "wordy" it has the propensity to become :upside_down_face:

**Comment by NightJar** [comment]
> Hey @NightJar looking good. I've got a just text changes to make this clearer.

Neat, thanks!

> **Text changes**
> 59318341-cf69e300-8d1a-11e9-8e02-de036fafe076-2

One suggestion about the first suggestion (changing "everyone" to "CMS acces") - currently the latter is entirely untrue. I'd like to think that if we were to have time to get around to the issue that deals with different user permissions as to who is required to have MFA and who isn't, then we'd make this change as part of that.

> Also, is there a reason why the radio labels are not aligned with radio?

Nope, hahaha just hadn't noticed! I think this might be default (i.e. `admin`) behaviour for radio buttons. I'll see if I can find another set to check against.
If it is, and no one else can replicate, then I'd be inclined to putting it down as "this is neither Windows nor Mac" and differences in font rendering engines.

File: client/src/legacy/SiteConfig.js (hunk 1 above)
Comment (NightJar) [reply client/src/legacy/SiteConfig.js]: Ah, you meant the scss rule :joy:

**Comment by NightJar** [comment]
OK, I've updated the text, save the one I pointed out above. (see #179 )

Also removed the now redundant rule from the css bundle :)

I've double checked that the text on the radio buttons is not unique to the MFA switch, and am marking that down as Linux font rendering related (see other radio buttons have the same issue in the expanded screenshot below).

image

File: src/Extension/SiteConfigExtension.php [hunk 3]
```diff
@@ -43,30 +41,27 @@ public function updateCMSFields(FieldList $fields)
...
-
             ]
         );
-        $mfaOptions->addExtraClass('mfa-settings__required mfa-settings--hidden');
+        $mfaOptions->addExtraClass('mfa-settings__required');
 
         $mfaGraceEnd = DateField::create(
             'MFAGracePeriodExpires',
-            'Optional: grace period end date when MFA is enforced'
+            'MFA will be required from (optional)'
```
Comment (robbieaverill) [inline src/Extension/SiteConfigExtension.php]: Needs translation

File: src/Extension/SiteConfigExtension.php [hunk 4]
```diff
@@ -91,7 +86,7 @@ protected function getHelpLink()
         return sprintf(
             '<a class="d-block mfa-settings__help-link" target="blank" rel="noopener" href="%s">%s</a>',
             $link,
-            _t(__CLASS__ . '.FIND_OUT_MORE', 'Find out more')
+            _t(__CLASS__ . '.FIND_OUT_MORE', 'Learn about MFA')
```
Comment (robbieaverill) [inline src/Extension/SiteConfigExtension.php]: This has a different purpose to the translation key now, maybe rename the key as well

**Pull request merged**